ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+"""
+All the flexy related classes and functionality lives here
+"""
+import logging
+import os
+import sys
+
+import io
+import configparser
+import subprocess
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework import config, merge_dict
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    get_ocp_version, TimeoutSampler, clone_repo
+)
+from ocs_ci.ocs import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+class FlexyBase(object):
+    """
+    A base class for all types of flexy installs
+    """
+    def __init__(self):
+        # Host dir path which will be mounted inside flexy container
+        # This will be root for all flexy housekeeping
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        self.flexy_mnt_host_dir = os.path.expanduser(
+            config.ENV_DATA['flexy_mnt_host_dir']
+        )
+        # Host dir path for private_conf dir
+        self.flexy_host_private_conf_dir_path = os.path.join(
+            self.flexy_mnt_host_dir, config.ENV_DATA['flexy_host_private_conf_dir']
+        )
+        # Path inside container where flexy_mnt_host_dir will be mounted
+        self.flexy_mnt_container_dir = config.ENV_DATA.get(
+            'flexy_mnt_container_dir', constants.FLEXY_MNT_CONTAINER_DIR
+        )
+        # Path inside container where flexy_private_conf_dir will be mounted
+        # same value needs to be set for BUSHSLICER_PRIVATE_DIR in env file
+        # set BUSHSLICER_PRIVATE_DIR in ocs-osp.env itself
+        self.flexy_mnt_private_conf_dir = config.ENV_DATA.get(
+            'flexy_mnt_private_conf_dir', constants.FLEXY_MNT_PRIVATE_CONF_DIR
+        )
+        self.flexy_openshift_misc_url = config.ENV_DATA.get(
+            'flexy_openshift_misc_url', constants.OCP_QE_MISC_REPO
+        )
+        self.flexy_private_conf_url = config.ENV_DATA.get(
+            'flexy_private_conf_url', constants.FLEXY_PRIVATE_CONF_URL
+        )
+        self.flexy_img_url = config.ENV_DATA.get('flexy_img_url')
+        self.flex_env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
```

Comment:
I think it might be worth to make the env file configurable and use `constants.FLEXY_DEFAULT_ENV_FILE` (pointing to `ocs-osp.env`) only as default value. For some configurations it might be worth to prepare new env file, instead of update all the required keys through another configuration file, WDYT?

Reply:
ok, this scenario I didn't thought about. We can make it configurable. I will try this one, because if we pass env file in that case we might have to skip cloning of the ocs-private repo all together. I will update you how it works out.

Reply:
@dahorak ,
How do you want this config file to be passed ? through --ocsci-conf ?

Reply:
I would use some configuration variable in configuration (e.g. `flexy_env_file` in `ENV_DATA` section, or somewhere else?).

Reply:
done, through CLI option ```--flexy-env-file```